nodestream_plugin_neptune/query_executor.py

```diff
@@ -0,0 +1,160 @@
+import asyncio
+import json
+import math
+from logging import getLogger
+from typing import Iterable
+
+from aiobotocore.session import get_session
+from nodestream.databases.query_executor import (
+    OperationOnNodeIdentity, OperationOnRelationshipIdentity, QueryExecutor)
+from nodestream.model import (IngestionHook, Node, RelationshipWithNodes,
+                              TimeToLiveConfiguration)
+
+from .ingest_query_builder import NeptuneDBIngestQueryBuilder
+from .query import Query, QueryBatch
+
+
+class NeptuneDBQueryExecutor(QueryExecutor):
+    def __init__(
+        self,
+        region,
+        host,
+        ingest_query_builder: NeptuneDBIngestQueryBuilder,
+        async_partitions=50,
+    ) -> None:
+        self.session = get_session()
+        self.region = region
+        self.host = host
+        self.ingest_query_builder = ingest_query_builder
+        self.logger = getLogger(self.__class__.__name__)
+        self.async_partitions = async_partitions
+
+    async def upsert_nodes_in_bulk_with_same_operation(
+        self, operation: OperationOnNodeIdentity, nodes: Iterable[Node]
+    ):
+        batched_query = (
+            self.ingest_query_builder.generate_batch_update_node_operation_batch(
+                operation, nodes
+            )
+        )
+        await self.execute_batch(batched_query)
+
+    async def upsert_relationships_in_bulk_of_same_operation(
+        self,
+        shape: OperationOnRelationshipIdentity,
+        relationships: Iterable[RelationshipWithNodes],
+    ):
+        queries = (
+            self.ingest_query_builder.generate_batch_update_relationship_query_batch(
+                shape, relationships
+            )
+        )
+        await self.execute_batch(queries)
+
+    async def perform_ttl_op(self, config: TimeToLiveConfiguration):
+        query = self.ingest_query_builder.generate_ttl_query_from_configuration(config)
+        await self.execute(query)
+
+    async def execute_hook(self, hook: IngestionHook):
+        query_string, params = hook.as_cypher_query_and_parameters()
+        await self.execute(Query(query_string, params))
+
+    def _split_parameters(self, parameters: list):
```

Comment:
The code between Neptune and Neptune Analytics is copied and pasted. Is it possible to refactor to reuse the code and avoid code duplication?